lib/spack/docs/build_settings.rst

```diff
@@ -179,15 +179,72 @@ Specific limitations include:
 
 * A package must define ``executables`` and ``determine_spec_details``
   for Spack to locate instances of that package.
-* This is currently intended to find build dependencies rather than
-  library packages.
+* This implementation only collects and examines executable files, so it
+  is only useful for build/run dependencies.
+* The logic does not search through modules module files, it can only
+  detect packages with executables defined in ``PATH``
 * Spack does not overwrite existing entries in the package configuration:
   If there is an external defined for a spec at any configuration scope,
   then Spack will not add a new external entry (``spack config blame packages``
   can help locate all external entries).
-* Currently this logic is focused on examining ``PATH`` and does not
-  search through modules (although it should find the package if a
-  module is loaded for it).
+
+As mentioned above, to make a package discoverable with
```

Comment:
IMO the stuff in the first bullet above and the details below really belong in the packaging guide.  `build_settings.rst` is about configuration -- not python code.  The packaging is about how to write packages.  Because they are related, I would add a note here that says something like "`spack external find` can only discover packages that support it -- for details on how to enable packages for external discovery, see <external discovery>" with a link to the section in the packaging guide.